### ISSUE TITLE
Timeout waiting to lock journal cache 

### ISSUE THREAD

**Issue by d1ss0nanz** [gradle/gradle#8750]
Running parallel gradle builds on Jenkins causes "Timeout waiting to lock journal cache" every now and then.

Notes:
We have not seen this behavior before switching to Gradle 5.
Disabling the Gradle daemon will not mitigate the issue.

### Expected Behavior
When running gradle builds in parallel, no lock timeout should occur.

### Current Behavior
```
Caused by: org.gradle.cache.LockTimeoutException: Timeout waiting to lock journal cache (/srv/jenkins/.gradle/caches/journal-1). It is currently in use by another Gradle instance.
Owner PID: 31422
Our PID: 3410
Owner Operation: 
Our operation: 
Lock file: /srv/jenkins/.gradle/caches/journal-1/journal-1.lock
	at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.lock(DefaultFileLockManager.java:291)
	at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.<init>(DefaultFileLockManager.java:161)
	at org.gradle.cache.internal.DefaultFileLockManager.lock(DefaultFileLockManager.java:107)
	at org.gradle.cache.internal.LockOnDemandCrossProcessCacheAccess.incrementLockCount(LockOnDemandCrossProcessCacheAccess.java:106)
	at org.gradle.cache.internal.LockOnDemandCrossProcessCacheAccess.acquireFileLock(LockOnDemandCrossProcessCacheAccess.java:168)
	at org.gradle.cache.internal.CrossProcessSynchronizingCache.put(CrossProcessSynchronizingCache.java:63)
	at org.gradle.api.internal.changedetection.state.DefaultFileAccessTimeJournal.setLastAccessTime(DefaultFileAccessTimeJournal.java:89)
	at org.gradle.internal.resource.local.SingleDepthFileAccessTracker.markAccessed(SingleDepthFileAccessTracker.java:60)
	at org.gradle.internal.resource.local.SingleDepthFileAccessTracker.markAccessed(SingleDepthFileAccessTracker.java:48)
	at org.gradle.internal.resource.cached.AbstractCachedIndex.lookup(AbstractCachedIndex.java:74)
	at org.gradle.api.internal.artifacts.ivyservice.modulecache.artifacts.DefaultModuleArtifactCache.lookup(DefaultModuleArtifactCache.java:77)
	at org.gradle.api.internal.artifacts.ivyservice.modulecache.artifacts.InMemoryModuleArtifactCache.lookup(InMemoryModuleArtifactCache.java:63)
	at org.gradle.api.internal.artifacts.ivyservice.ivyresolve.CachingModuleComponentRepository$LocateInCacheRepositoryAccess.resolveArtifactFromCache(CachingModuleComponentRepository.java:323)
	at org.gradle.api.internal.artifacts.ivyservice.ivyresolve.CachingModuleComponentRepository$LocateInCacheRepositoryAccess.resolveArtifact(CachingModuleComponentRepository.java:290)
	at org.gradle.api.internal.artifacts.ivyservice.ivyresolve.ErrorHandlingModuleComponentRepository$ErrorHandlingModuleComponentRepositoryAccess.lambda$resolveArtifact$12(ErrorHandlingModuleComponentRepository.java:186)
	at org.gradle.api.internal.artifacts.ivyservice.ivyresolve.ErrorHandlingModuleComponentRepository$ErrorHandlingModuleComponentRepositoryAccess.tryResolveAndMaybeBlacklist(ErrorHandlingModuleComponentRepository.java:242)
	... 245 more
```
### Context

- Using gradle wrapper
- gradle version 5.2.1

### Steps to Reproduce (for bugs)
Run builds in parallel on the same machine.

**Comment by Kipre** [comment]
Can also happen when running multiple builds in a row.

**Comment by sedovalx** [comment]
In my case, the owner of the lock is MacOS `uninstalld` process. And that is quite strange...

**Comment by guilherme-pereira-ifood** [comment]
@d1ss0nanz could you solve it somehow?

**Comment by dajudge** [comment]
I'm running into the same issue - is there a known solution to this?

**Comment by FabianSawatzki1234** [comment]
+1

**Comment by d1ss0nanz** [comment]
We did not find a solution.

But I've moved on from the project I was using Gradle with in the meantime.

**Comment by riklopfer** [comment]
Our solution is to specify a different `gradle-user-home` for each jobs. Specifically, setting it to the current working directory (i.e. project directory). 

```
./gradlew -g $PWD buildOrWhatever
```

**Comment by asarkar** [comment]
@riklopfer What is the implication of your suggestion?

**Comment by Alexminator99** [comment]
find ~/.gradle -type f -name "*.lock" | while read f; do rm $f; done this works for some people... for me dont... so if anyone have another solutions please share

**Comment by martinda** [comment]
It is the same problem with user accounts mounted via NFS to multiple hosts.

**Comment by DavidDTA** [comment]
We are running into this as well.

**Comment by fembr1** [comment]
Same here

**Comment by ph4r05** [comment]
tldr: In my case the culprit was restrictive firewall, part of the antivirus system.

I was struggling with the same problem for hours. 

According to the https://discuss.gradle.org/t/timeout-waiting-to-lock-file-hash-cache/21961/2 and https://github.com/elastic/elasticsearch/issues/50954#issuecomment-576772154 , gradle server is keeping the lock and releases it on request of other gradle process. The communication goes over UDP socket on localhost. Thus if there is a firewall blocking this communication, gradle keeps file locks and other gradle processes cannot continue.

Try disabling AV / firewall to see if problem vanishes. Running gradle with `--debug` provides you PID of the lock owner.

**Comment by ilor** [comment]
I've been hitting this too, and using separate gradle home directories per https://github.com/gradle/gradle/issues/8750#issuecomment-605016788 mitigated the issue. It's a bit annoying due to disk space usage, though.

**Comment by GooDiVer** [comment]
I advice you try to kill all gradle daemons. Works for me.
Execute  in command line: `./gradlew --stop`

**Comment by jbartok** [comment]
I'm experiencing this issue when there are Gradle Daemons of different versions running in parallel on my machine, in my case `6.8.3` and `7.2`.

**Comment by Amokrane** [comment]
I just ran into this on my machine, solved it by first checking which process was currently holding that lock, then killed it: 

```
PID=fuser -c <file_to_journal_lock_file>
kill -9 $PID
```

**Comment by jbartok** [comment]
I have finally managed to figure out the root cause on my machine. Basically the first daemon started on the machine holds such global locks until some other daemon REQUESTS it from them, then they hand it over to the requester. This requesting process happens through the local network, via UDP. What happened on my machine was that my Antivirus was blocking this communication so second daemon wasn't able to get its request to the one holding the lock. Disabling the Antivirus (later re-enabling, but adding an exception) solved the problem for good.

**Comment by injecteer** [comment]
@jbartok Can you elaborate on what you did to your AV? What exception have you added?

I disabled all FWs, turned the WinDefender off:
![image](https://user-images.githubusercontent.com/32952/133082413-a7db54d6-a45d-49fd-9ac2-fa7e5455d781.png)
![image](https://user-images.githubusercontent.com/32952/133082463-2c8c463b-4c11-4f2b-bc48-4dca90c4fda1.png)

 but still getting file locks exceptions:

```
Gradle could not start your build.
> Could not create service of type FileAccessTimeJournal using GradleUserHomeScopeServices.createFileAccessTimeJournal().
   > Timeout waiting to lock journal cache (C:\Users\injec\.gradle\caches\journal-1). It is currently in use by another Gradle instance.
     Owner PID: 1912
     Our PID: 6676
     Owner Operation: 
     Our operation: 
     Lock file: C:\Users\injec\.gradle\caches\journal-1\journal-1.lock
```
further details in [this SO Question](https://stackoverflow.com/questions/69087223/gradle-daemon-stopped-working-in-parallel-due-to-file-lock-timeouts)

**Comment by jbartok** [comment]
I'm actually on a Mac, using ESET Cyber Security Pro. Shutting it down, or adding a rule to allow all connections from localhost to localhost is what fixed the issue for me. Have not had any problems since. Basically my Gradle daemons were stuck while trying to contact the first daemon holding the lock (in order to request ownership from it).

**Comment by primeapple** [comment]
> I'm actually on a Mac, using ESET Cyber Security Pro. Shutting it down, or adding a rule to allow all connections from localhost to localhost is what fixed the issue for me. Have not had any problems since. Basically my Gradle daemons were stuck while trying to contact the first daemon holding the lock (in order to request ownership from it).

I had the same issue and found that disabling the `Web and EMail protection` of ESET helped, when disabling the firewall did not bring any succes.

**Comment by kibotu** [comment]
> Our solution is to specify a different `gradle-user-home` for each job. Specifically, setting it to the current working directory (i.e. project directory).
> 
> ```
> ./gradlew -g $PWD buildOrWhatever
> ```


we add this to our ci scripts

`sh "echo systemProp.gradle.user.home=./gradleHome >> gradle.properties"`

which changes the gradle home folder for each job, it kind of defeats the purpose of gradle caches but better that than having flaky ci builds :/

**Comment by kkalisz** [comment]
When I moved to version 7.x of Gradle I have this issue all the time. And Yes I can confirm that I'm working on few projects with different Gradle versions at the same time. But just want to notice that before version 7 I never encountered that issue.

**Comment by nickcaballero** [comment]
Encountered this issue when running multiple Gradle builds inside Docker containers. Running as a daemon and using host network in Docker resolved the issue  -https://github.com/gradle/gradle/issues/8750#issuecomment-915312188.

Using the host network in Docker makes it possible for the daemons to talk to each other.

**Comment by cjbrooks12** [comment]
I've been running into this issue with Netskope as the firewall, here's the relevant info to reproduce on my machine:

- MacBook Pro running macOS 11.5.2 (Big Sur)
- Gradle being run in parallel with multiple JDKs
  - First start running normal Gradle "run" task, when terminal JAVA_HOME is set to one JDK (11)
  - After run task starts its Gradle Daemon, run IntelliJ Gradle sync when IJ project is configured with another JDK (8)
- Project is using Gradle wrapper, version 7.2
- Netskope is the culprit. When Netskope is disabled, the problem goes away.
  - Neither my company's IT team nor Netskope support can figure out any particular configuration rules that should block this communication.
  - We also could not find any suspicious activity when viewing local or network traffic with Wireshark.

Additionally, the Netskope support team would be interested in engaging with Gradle engineers to figure out what exactly is going on here, if possible.

**Comment by jbartok** [comment]
> I've been running into this issue with Netskope as the firewall, here's the relevant info to reproduce on my machine:
> 
> * MacBook Pro running macOS 11.5.2 (Big Sur)
> * Gradle being run in parallel with multiple JDKs
>   
>   * First start running normal Gradle "run" task, when terminal JAVA_HOME is set to one JDK (11)
>   * After run task starts its Gradle Daemon, run IntelliJ Gradle sync when IJ project is configured with another JDK (8)
> * Project is using Gradle wrapper, version 7.2
> * Netskope is the culprit. When Netskope is disabled, the problem goes away.
>   
>   * Neither my company's IT team nor Netskope support can figure out any particular configuration rules that should block this communication.
>   * We also could not find any suspicious activity when viewing local or network traffic with Wireshark.
> 
> Additionally, the Netskope support team would be interested in engaging with Gradle engineers to figure out what exactly is going on here, if possible.

[FileLockCommunicator](https://github.com/gradle/gradle/blob/master/subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/locklistener/FileLockCommunicator.java) is binding a UDP socket to the wildcard address, in order to be receiving broadcasts. Maybe this is the behavior that's being blocked by Netskope. Maybe it disallows broadcasting by default.

**Comment by Xnyle** [comment]
+1 after moving to gradle 7
Having to do complex network setups or poke holes in k8s / docker container isolation is IMHO the wrong approach.
Gradle itself should provide a straight forward approach where you can a) have a global cache that you can mount into build containers in order to speed up things b) use that cache concurrently without having to worry about inter container communication which is hard to setup.

If gradle has to synchronize different accessors to the cache that should be done with short time locks, not by locking a journal for the whole timespan of the execution.

**Comment by athrunsaga** [comment]
> tldr: In my case the culprit was restrictive firewall, part of the antivirus system.
> 
> According to the https://discuss.gradle.org/t/timeout-waiting-to-lock-file-hash-cache/21961/2 and [elastic/elasticsearch#50954 (comment)](https://github.com/elastic/elasticsearch/issues/50954#issuecomment-576772154), [SO](https://stackoverflow.com/questions/21523508/it-is-currently-in-use-by-another-gradle-instance/67624525#67624525), gradle server is keeping the lock and releases it on request of other gradle process. The communication goes over UDP socket on localhost. Thus if there is a firewall blocking this communication, gradle keeps file locks and other gradle processes cannot continue.
> 
> Try disabling AV / firewall to see if problem vanishes. Running gradle with `--debug` provides you PID of the lock owner.

I tried the @ph4r05 method and it worked for me.

**Comment by cocorossello** [comment]
Another workaround is making the build cache directory local in jenkins, combined with a remote cache it gives a decent performance.

```
buildCache {
    local {
        directory = new File(rootDir, '.gradle/build-cache')
        removeUnusedEntriesAfterDays = 5
    }
    remote(HttpBuildCache) {
        ...
```

**Comment by sergey-morenets** [comment]
The same issue happened in our project where multiple sub-projects are built inside Docker (Gradle 7.4.2):

```
To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/7.4.2/userguide/gradle_daemon.html#sec:disabling_the_daemon.
Daemon will be stopped at the end of the build
Configuration on demand is an incubating feature.                                                                                                                          
FAILURE: Build failed with an exception.
What went wrong:
Gradle could not start your build.
> Cannot create service of type BuildTreeActionExecutor using method LauncherServices$ToolingBuildTreeScopeServices.createActionExecutor() as there is a problem with parameter #1 of type List<BuildActionRunner>.
 > Cannot create service of type BuildModelActionRunner using BuildModelActionRunner constructor as there is a problem with parameter #1 of type PayloadSerializer.
 > Cannot create service of type PayloadSerializer using method LauncherServices$ToolingGradleUserHomeScopeServices.createPayloadSerializer() as there is a problem with parameter #2 of type PayloadClassLoaderFactory.
> Cannot create service of type PayloadClassLoaderFactory using method LauncherServices$ToolingGradleUserHomeScopeServices.createClassLoaderFactory() as there is a problem with parameter #1 of type CachedClasspathTransformer.
 > Cannot create service of type DefaultCachedClasspathTransformer using DefaultCachedClasspathTransformer constructor as there is a problem with parameter #6 of type FileSystemAccess.
 > Cannot create service of type FileSystemAccess using method VirtualFileSystemServices$GradleUserHomeServices.createFileSystemAccess() as there is a problem with parameter #2 of type VirtualFileSystem.
 > Cannot create service of type BuildLifecycleAwareVirtualFileSystem using method VirtualFileSystemServices$GradleUserHomeServices.createVirtualFileSystem() as there is a problem with parameter #6 of type GlobalCacheLocations.
 > Cannot create service of type GlobalCacheLocations using method GradleUserHomeScopeServices.createGlobalCacheLocations() as there is a problem with parameter #1 of type List<GlobalCache>.
 > Could not create service of type FileAccessTimeJournal using GradleUserHomeScopeServices.createFileAccessTimeJournal().
 > Timeout waiting to lock journal cache (/home/gradle/.gradle/caches/journal-1). It is currently in use by another Gradle instance.
 Owner PID: 47
 Our PID: 47 
 Owner Operation:
 Our operation:
Lock file: /home/gradle/.gradle/caches/journal-1/journal-1.lock
```

**Comment by agascon** [comment]
Hi @cocorossello, would you mind to share details about your local + remote buildCache setup?

We are facing this issue in a CI environment due to several jobs running at the same time, and your approach sounds like a great solution to me.

Thanks!

**Comment by codingsoul** [comment]
I am using an exFat formatted external drive with M1 macbook pro & my issue got fixed when i downgraded my Android Studio gradle plugin version from 7.1.2 to 7.0.0 for some reason.

**Comment by JGonczoroski** [comment]
I have the similar issue. Every time I run the application, this occurs. When I delete the PID, I can run it once, but if I re-run, it happens again.

**Comment by tplive** [comment]
We had this issue after upgrading Gradle to 7.4 and AGP to 7.3.1 using the AGP Upgrade Assistant. We tried everything above, but it seemed it was impossible to turn off the gradle cache. So in the end what solved it for us, was upgrading the Gradle Wrapper script, using this method:

`./gradlew wrapper --gradle-version 7.4`

Now the build is running green, with cache and all...

**Comment by aSemy** [comment]
> > Our solution is to specify a different `gradle-user-home` for each job. Specifically, setting it to the current working directory (i.e. project directory).
> > ```
> > ./gradlew -g $PWD buildOrWhatever
> > ```
> 
> we add this to our ci scripts
> 
> `sh "echo systemProp.gradle.user.home=./gradleHome >> gradle.properties"`
> 
> which changes the gradle home folder for each job, it kind of defeats the purpose of gradle caches but better that than having flaky ci builds :/

As well as overriding `gradle.user.home` to be a distinct directory you could additionally specify the original directory as a cache source with [`GRADLE_RO_DEP_CACHE`](https://docs.gradle.org/current/userguide/dependency_resolution.html#sub:shared-readonly-cache), e.g. `GRADLE_RO_DEP_CACHE=$HOME/.gradle/caches/`

**Comment by injecteer** [comment]
>  C:\Users<me>.gradle\caches. 
After addeing the exclusion I could also start 3 Gradle apps in parallel. Great!

**Comment by lwshuo** [comment]
We are seeing this issue with the same Owner PID and Our PID
```

Gradle could not start your build.
> Cannot create service of type BuildTreeActionExecutor using method LauncherServices$ToolingBuildTreeScopeServices.createActionExecutor() as there is a problem with parameter #1 of type List<BuildActionRunner>.
   > Cannot create service of type BuildModelActionRunner using BuildModelActionRunner constructor as there is a problem with parameter #1 of type PayloadSerializer.
      > Cannot create service of type PayloadSerializer using method LauncherServices$ToolingGradleUserHomeScopeServices.createPayloadSerializer() as there is a problem with parameter #2 of type PayloadClassLoaderFactory.
         > Cannot create service of type PayloadClassLoaderFactory using method LauncherServices$ToolingGradleUserHomeScopeServices.createClassLoaderFactory() as there is a problem with parameter #1 of type CachedClasspathTransformer.
            > Cannot create service of type DefaultCachedClasspathTransformer using DefaultCachedClasspathTransformer constructor as there is a problem with parameter #6 of type FileSystemAccess.
               > Cannot create service of type FileSystemAccess using method VirtualFileSystemServices$GradleUserHomeServices.createFileSystemAccess() as there is a problem with parameter #2 of type VirtualFileSystem.
                  > Cannot create service of type BuildLifecycleAwareVirtualFileSystem using method VirtualFileSystemServices$GradleUserHomeServices.createVirtualFileSystem() as there is a problem with parameter #6 of type GlobalCacheLocations.
                     > Cannot create service of type GlobalCacheLocations using method GradleUserHomeScopeServices.createGlobalCacheLocations() as there is a problem with parameter #1 of type List<GlobalCache>.
                        > Could not create service of type FileAccessTimeJournal using GradleUserHomeScopeServices.createFileAccessTimeJournal().
                           > Timeout waiting to lock journal cache (/drone/src/.gradle-cache/caches/journal-1). It is currently in use by another Gradle instance.
                             Owner PID: 44
                             Our PID: 44
                             Owner Operation: 
                             Our operation: 
                             Lock file: /drone/src/.gradle-cache/caches/journal-1/journal-1.lock
```

**Comment by shaikezr** [comment]
> We are seeing this issue with the same Owner PID and Our PID

Had the same issue as @lwshuo pop up for me today. Running `./gradlew --stop` resolved for me.

**Comment by trajano** [comment]
What I did was exclude the journal folder in my Dockerfile multistage build

```Dockerfile
FROM gradle:7 AS deps
WORKDIR /work
COPY ./*.gradle /work
COPY ./gradle/*.gradle /work/gradle/
RUN --mount=type=secret,id=init-gradle \
    --mount=type=cache,target=/root/.gradle/caches/jars-9 \
    --mount=type=cache,target=/root/.gradle/caches/modules-2 \
  mkdir -p /root/.gradle &&\
  cp /run/secrets/init-gradle /root/.gradle/init.gradle && \
  gradle -q --no-daemon dependencies && \
  rm /root/.gradle/init.gradle
COPY . /work/

FROM deps AS build
RUN --mount=type=secret,id=init-gradle \
    --mount=type=cache,target=/root/.gradle/caches/jars-9 \
    --mount=type=cache,target=/root/.gradle/caches/modules-2 \
  mkdir -p /root/.gradle && \
  cp /run/secrets/init-gradle /root/.gradle/init.gradle && \
  gradle -q build --no-daemon -x test && \
  rm /root/.gradle/init.gradle

```

**Comment by alloc64** [comment]
For anyone having this issue while building multiple docker containers in parallel, setting gradle cache (~/.gradle) sharing to locked fixes the issue. But for the cost of disabled parallel docker builds.

```dockerfile
FROM gradle:7.5.1-jdk17-focal AS build
COPY --chown=gradle:gradle . /work
WORKDIR /work
RUN --mount=type=cache,sharing=locked,target=/root/.gradle gradle clean build --no-daemon -x test --parallel
```

**Comment by cpdeethree** [comment]
I'm not sure this is entirely without its own pitfalls yet, but one thing that we've done that resolved the issue for multiple docker containers running in parallel and sharing the same cache is the following:

Create a shared cache directory, let's call it ~/gradle-cache. This directory should be read/write locked so that only one operation can be writing to it at a time. Mount that into the container.

Before calling gradle, rsync that to ~/.gradle in the container

Run gradle. Local copy of ~/.gradle is used during the run

After gradle is finished, rsync the updated folder back to ~/gradle-cache

**Comment by trajano** [comment]
> I'm not sure this is entirely without its own pitfalls yet, but one thing that we've done that resolved the issue for multiple docker containers running in parallel and sharing the same cache is the following:
> 
> Create a shared cache directory, let's call it ~/gradle-cache. This directory should be read/write locked so that only one operation can be writing to it at a time. Mount that into the container.
> 
> Before calling gradle, rsync that to ~/.gradle in the container
> 
> Run gradle. Local copy of ~/.gradle is used during the run
> 
> After gradle is finished, rsync the updated folder back to ~/gradle-cache

Main pitfall I can see is if you have `init.gradle` with your private repo settings it may be put into the cache.

**Comment by trajano** [comment]
> For anyone having this issue while building multiple docker containers in parallel, setting gradle cache (~/.gradle) sharing to locked fixes the issue. But for the cost of disabled parallel docker builds.
> 
> ```dockerfile
> FROM gradle:7.5.1-jdk17-focal AS build
> COPY --chown=gradle:gradle . /work
> WORKDIR /work
> RUN --mount=type=cache,sharing=locked,target=/root/.gradle gradle clean build --no-daemon -x test --parallel
> ```

where did you see that sharing=locked documented?

**Comment by landsman** [comment]
It just happened to me on localhost, even without docker.

**Comment by lost22git** [comment]
same problem ;-(

**Comment by 95jinhong** [comment]
hello, i am having the same problem as above.
I am using Action Runner as a Pod in a Kubernetes environment.

I am using a .gradle folder shared by each runner via NFS, and I am using Readonly Cache individually using the "GRADLE_RO_DEP_CACHE" option.

A few times the runner runs fine, but sometimes the runner fails.
```
[success-log]
Run ./gradlew clean app:assembleDebug --no-daemon
To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/7.5.1/userguide/gradle_daemon.html#sec:disabling_the_daemon.
Daemon will be stopped at the end of the build 
Shared read-only dependency cache is an incubating feature.

[failed-log]
Run ./gradlew domain:test --no-daemon
To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/7.5.1/userguide/gradle_daemon.html#sec:disabling_the_daemon.
Daemon will be stopped at the end of the build 

FAILURE: Build failed with an exception.

* What went wrong:
Gradle could not start your build.
> Cannot create service of type BuildSessionActionExecutor using method LauncherServices$ToolingBuildSessionScopeServices.createActionExecutor() as there is a problem with parameter #21 of type FileSystemWatchingInformation.
   > Cannot create service of type BuildLifecycleAwareVirtualFileSystem using method VirtualFileSystemServices$GradleUserHomeServices.createVirtualFileSystem() as there is a problem with parameter #7 of type GlobalCacheLocations.
      > Cannot create service of type GlobalCacheLocations using method GradleUserHomeScopeServices.createGlobalCacheLocations() as there is a problem with parameter #1 of type List<GlobalCache>.
         > Could not create service of type FileAccessTimeJournal using GradleUserHomeScopeServices.createFileAccessTimeJournal().
            > Timeout waiting to lock journal cache (/mnt/.gradle/caches/journal-1). It is currently in use by another Gradle instance.
              Owner PID: 52[4](https://github.com/boostbrothers/ddocdoc-android/actions/runs/7382550848/job/20082537320#step:6:5)
              Our PID: 3[10](https://github.com/boostbrothers/ddocdoc-android/actions/runs/7382550848/job/20082537320#step:6:11)
              Owner Operation: 
              Our operation: 
              Lock file: /mnt/.gradle/caches/journal-1/journal-1.lock
```

Is there any way to fix this?

**Comment by AvishayHirsh** [comment]
any update about this issue ? is it on roadmap to be fixed ?

**Comment by baksha97** [comment]
> According to the https://discuss.gradle.org/t/timeout-waiting-to-lock-file-hash-cache/21961/2 and [elastic/elasticsearch#50954 (comment)](https://github.com/elastic/elasticsearch/issues/50954#issuecomment-576772154), [SO](https://stackoverflow.com/questions/21523508/it-is-currently-in-use-by-another-gradle-instance/67624525#67624525), gradle server is keeping the lock and releases it on request of other gradle process. The communication goes over UDP socket on localhost. Thus if there is a firewall blocking this communication, gradle keeps file locks and other gradle processes cannot continue.

That's interesting. I've been seeing this often on my self-hosted macOS runners.

This might help identify the issues on build systems if we check before each job.
```bash
{ echo "test message" | nc -u 127.0.0.1 9999 & sleep 0.1; } && nc -ul 127.0.0.1 9999 -w 1 | grep "test message" && echo "UDP message successfully received on loopback address" || echo "Failed to receive UDP message on loopback address"
```

**Comment by shalva97** [comment]
Any updates here. I have same issue. maybe it is a good idea to lower the timeout? currently it looks like it is 1 minute. if this was 2-3 seconds I would save a bit more time

```

FAILURE: Build failed with an exception.

* What went wrong:
Timeout waiting to lock jars (/Users/Giorgi_Shalvashvili/.gradle/caches/jars-9). It is currently in use by another Gradle instance.
Owner PID: 17789
Our PID: 33961
Owner Operation: 
Our operation: 
Lock file: /Users/Giorgi_Shalvashvili/.gradle/caches/jars-9/jars-9.lock

* Try:
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

* Exception is:
org.gradle.cache.LockTimeoutException: Timeout waiting to lock jars (/Users/Giorgi_Shalvashvili/.gradle/caches/jars-9). It is currently in use by another Gradle instance.
Owner PID: 17789
Our PID: 33961
Owner Operation: 
Our operation: 
Lock file: /Users/Giorgi_Shalvashvili/.gradle/caches/jars-9/jars-9.lock
	at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.timeoutException(DefaultFileLockManager.java:374)
	at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.lock(DefaultFileLockManager.java:335)
	at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.<init>(DefaultFileLockManager.java:179)
	at org.gradle.cache.internal.DefaultFileLockManager.lock(DefaultFileLockManager.java:125)
	at org.gradle.cache.internal.LockOnDemandCrossProcessCacheAccess.incrementLockCount(LockOnDemandCrossProcessCacheAccess.java:106)
	at org.gradle.cache.internal.LockOnDemandCrossProcessCacheAccess.acquireFileLock(LockOnDemandCrossProcessCacheAccess.java:163)
	at org.gradle.cache.internal.DefaultCacheCoordinator.onStartWork(DefaultCacheCoordinator.java:401)
	at org.gradle.cache.internal.DefaultCacheCoordinator.useCache(DefaultCacheCoordinator.java:240)
	at org.gradle.cache.internal.DefaultPersistentDirectoryStore.useCache(DefaultPersistentDirectoryStore.java:159)
	at org.gradle.cache.internal.DefaultCacheFactory$ReferenceTrackingCache.useCache(DefaultCacheFactory.java:216)
	at org.gradle.internal.classpath.DefaultCachedClasspathTransformer$ParallelTransformExecutor.transformAll(DefaultCachedClasspathTransformer.java:226)
	at org.gradle.internal.classpath.DefaultCachedClasspathTransformer.copyingTransform(DefaultCachedClasspathTransformer.java:112)
	at org.gradle.internal.classpath.DefaultCachedClasspathTransformer.copyingTransform(DefaultCachedClasspathTransformer.java:58)
	at org.gradle.internal.daemon.serialization.DaemonSidePayloadClassLoaderFactory.createClassLoaderForClassPath(DaemonSidePayloadClassLoaderFactory.java:73)
	at org.gradle.internal.daemon.serialization.DaemonSidePayloadClassLoaderFactory.getClassLoaderFor(DaemonSidePayloadClassLoaderFactory.java:46)
	at org.gradle.tooling.internal.provider.serialization.DefaultPayloadClassLoaderRegistry$ClassLoaderToDetailsTransformer.transform(DefaultPayloadClassLoaderRegistry.java:181)
	at org.gradle.tooling.internal.provider.serialization.DefaultPayloadClassLoaderRegistry$ClassLoaderToDetailsTransformer.transform(DefaultPayloadClassLoaderRegistry.java:168)
	at org.gradle.tooling.internal.provider.serialization.ClassLoaderCache.getClassLoader(ClassLoaderCache.java:52)
	at org.gradle.tooling.internal.provider.serialization.DefaultPayloadClassLoaderRegistry.getClassLoader(DefaultPayloadClassLoaderRegistry.java:103)
	at org.gradle.tooling.internal.provider.serialization.DefaultPayloadClassLoaderRegistry.access$300(DefaultPayloadClassLoaderRegistry.java:47)
	at org.gradle.tooling.internal.provider.serialization.DefaultPayloadClassLoaderRegistry$2.resolveClass(DefaultPayloadClassLoaderRegistry.java:96)
	at org.gradle.tooling.internal.provider.serialization.WellKnownClassLoaderRegistry$2.resolveClass(WellKnownClassLoaderRegistry.java:91)
	at org.gradle.tooling.internal.provider.serialization.PayloadSerializerObjectInputStream.readClass(PayloadSerializerObjectInputStream.java:66)
	at org.gradle.tooling.internal.provider.serialization.PayloadSerializerObjectInputStream.readClassDescriptor(PayloadSerializerObjectInputStream.java:46)
	at org.gradle.tooling.internal.provider.serialization.PayloadSerializer.deserialize(PayloadSerializer.java:76)
	at org.gradle.tooling.internal.provider.runner.ClientProvidedPhasedActionRunner.run(ClientProvidedPhasedActionRunner.java:57)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.internal.buildtree.ProblemReportingBuildActionRunner.run(ProblemReportingBuildActionRunner.java:49)
	at org.gradle.launcher.exec.BuildOutcomeReportingBuildActionRunner.run(BuildOutcomeReportingBuildActionRunner.java:65)
	at org.gradle.tooling.internal.provider.FileSystemWatchingBuildActionRunner.run(FileSystemWatchingBuildActionRunner.java:140)
	at org.gradle.launcher.exec.BuildCompletionNotifyingBuildActionRunner.run(BuildCompletionNotifyingBuildActionRunner.java:41)
	at org.gradle.launcher.exec.RootBuildLifecycleBuildActionExecutor.lambda$execute$0(RootBuildLifecycleBuildActionExecutor.java:54)
	at org.gradle.composite.internal.DefaultRootBuildState.run(DefaultRootBuildState.java:130)
	at org.gradle.launcher.exec.RootBuildLifecycleBuildActionExecutor.execute(RootBuildLifecycleBuildActionExecutor.java:54)
	at org.gradle.internal.buildtree.InitDeprecationLoggingActionExecutor.execute(InitDeprecationLoggingActionExecutor.java:62)
	at org.gradle.internal.buildtree.InitProblems.execute(InitProblems.java:36)
	at org.gradle.internal.buildtree.DefaultBuildTreeContext.execute(DefaultBuildTreeContext.java:40)
	at org.gradle.launcher.exec.BuildTreeLifecycleBuildActionExecutor.lambda$execute$0(BuildTreeLifecycleBuildActionExecutor.java:71)
	at org.gradle.internal.buildtree.BuildTreeState.run(BuildTreeState.java:60)
	at org.gradle.launcher.exec.BuildTreeLifecycleBuildActionExecutor.execute(BuildTreeLifecycleBuildActionExecutor.java:71)
	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor$3.call(RunAsBuildOperationBuildActionExecutor.java:61)
	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor$3.call(RunAsBuildOperationBuildActionExecutor.java:57)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor.execute(RunAsBuildOperationBuildActionExecutor.java:57)
	at org.gradle.launcher.exec.RunAsWorkerThreadBuildActionExecutor.lambda$execute$0(RunAsWorkerThreadBuildActionExecutor.java:36)
	at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:263)
	at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:127)
	at org.gradle.launcher.exec.RunAsWorkerThreadBuildActionExecutor.execute(RunAsWorkerThreadBuildActionExecutor.java:36)
	at org.gradle.tooling.internal.provider.continuous.ContinuousBuildActionExecutor.execute(ContinuousBuildActionExecutor.java:110)
	at org.gradle.tooling.internal.provider.SubscribableBuildActionExecutor.execute(SubscribableBuildActionExecutor.java:64)
	at org.gradle.internal.session.DefaultBuildSessionContext.execute(DefaultBuildSessionContext.java:46)
	at org.gradle.internal.buildprocess.execution.BuildSessionLifecycleBuildActionExecutor$ActionImpl.apply(BuildSessionLifecycleBuildActionExecutor.java:92)
	at org.gradle.internal.buildprocess.execution.BuildSessionLifecycleBuildActionExecutor$ActionImpl.apply(BuildSessionLifecycleBuildActionExecutor.java:80)
	at org.gradle.internal.session.BuildSessionState.run(BuildSessionState.java:71)
	at org.gradle.internal.buildprocess.execution.BuildSessionLifecycleBuildActionExecutor.execute(BuildSessionLifecycleBuildActionExecutor.java:62)
	at org.gradle.internal.buildprocess.execution.BuildSessionLifecycleBuildActionExecutor.execute(BuildSessionLifecycleBuildActionExecutor.java:41)
	at org.gradle.internal.buildprocess.execution.StartParamsValidatingActionExecutor.execute(StartParamsValidatingActionExecutor.java:64)
	at org.gradle.internal.buildprocess.execution.StartParamsValidatingActionExecutor.execute(StartParamsValidatingActionExecutor.java:32)
	at org.gradle.internal.buildprocess.execution.SessionFailureReportingActionExecutor.execute(SessionFailureReportingActionExecutor.java:51)
	at org.gradle.internal.buildprocess.execution.SessionFailureReportingActionExecutor.execute(SessionFailureReportingActionExecutor.java:39)
	at org.gradle.internal.buildprocess.execution.SetupLoggingActionExecutor.execute(SetupLoggingActionExecutor.java:47)
	at org.gradle.internal.buildprocess.execution.SetupLoggingActionExecutor.execute(SetupLoggingActionExecutor.java:31)
	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:70)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:39)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:29)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:35)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.lambda$execute$0(ForwardClientInput.java:40)
	at org.gradle.internal.daemon.clientinput.ClientInputForwarder.forwardInput(ClientInputForwarder.java:80)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:64)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:63)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:84)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:52)
	at org.gradle.launcher.daemon.server.DaemonStateCoordinator.lambda$runCommand$0(DaemonStateCoordinator.java:321)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:48)


BUILD FAILED in 1m
```

**Comment by GabeK0** [comment]
Wanted to flag this somewhat related issue: https://github.com/gradle/gradle/issues/31245

People experiencing these timeouts more recently (around December 2024) could have the root cause be from firewall software + macOS 15, and disabling said firewall software can be a potential temporary workaround.

**Comment by ljacomet** [comment]
Thanks @GabeK0 for linking those.

Given what was discovered in the analysis of #31245, and the fixes made for it (#31964 and in particular #31953), it would be interesting to see if anyone having a setup where this issue did occur frequently could try with Gradle >= 8.12.1 and report.

Gradle should be more resilient to these kind of failures.

**Comment by cocorossello** [comment]
> Thanks [@GabeK0](https://github.com/GabeK0) for linking those issues.
> 
> Given what was discovered in the analysis of [#31245](https://github.com/gradle/gradle/issues/31245), and the fixes made for it ([#31964](https://github.com/gradle/gradle/pull/31964) and in particular [#31953](https://github.com/gradle/gradle/pull/31953)), it would be interesting to see if anyone having a setup where this issue did occur frequently could try with Gradle >= 8.12.1 and report.
> 
> Gradle should be more resilient to these kind of failures.

We had this problem and we did a workaround by  [not sharing the build cache directory](https://github.com/gradle/gradle/issues/8750#issuecomment-1092604100).

I removed the workaround and tried with 8.12.1 and we haven't seen the error in 2 days

**Comment by pedrolamarao** [comment]
I am working on an app built by Docker Compose where two different images are built by gradle sharing the cache via --mount=type=cache. Every now and then, this problem occurs.

**Comment by mlopatkin** [comment]
This issue will be closed as a duplicate of

* #8375

Please add your use case and 👍 to that issue.

If you think our analysis is wrong, please provide us with more detailed information explaining why.
